Fix performance issue on signups with many posts.

DFurnes opened the pull request:
### What's this PR do?

This pull request fixes a performance issue with the `Signup.quantity` field on posts with many signups. Because we updated this to store quantity on individual posts, the `$signup->posts` relationship was loading _all_ the posts on a signup into memory to sum them up:

![Screen Shot 2019-12-04 at 4 00 13 PM](https://user-images.githubusercontent.com/583202/70187631-b97f1e80-16bc-11ea-8ef3-d31bd73716bf.png)

Yikes! I adjusted this to create an "aggregate" query with the query builder rather than loading these all and using [Eloquent's Collection](https://laravel.com/docs/6.x/eloquent-collections) class to tally them up in-memory. Much better:

<img width="603" alt="Screen Shot 2019-12-04 at 5 43 54 PM" src="https://user-images.githubusercontent.com/583202/70188031-a751b000-16bd-11ea-8382-0984cbf6961b.png">

This also removes the `features.v3QuantitySupport` feature flag since it shipped a long time ago.

### How should this be reviewed?

👀 

### Any background context you want to provide?

This was causing intermittent failures for our Ghost Inspector user, which has 5,937 posts on their "Merit Over Money" signup. I'll make a backlog ticket to see if we can remove this `Signup.quantity` field altogether on the base response (and make it opt-in `?include=quantity` so we don't have to run this extra query unnecessarily).

### Relevant tickets

References [Pivotal #170107177](https://www.pivotaltracker.com/story/show/170107177).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.

DFurnes commented:
It may not be worth doing right now, but we should also think about whether we can delete the `quantity` and `quantity_pending` columns on the `signups` table (since they're no longer used).

@katiecrane Since you were around when we made the switch, any reasons you can think not to?

aaronschachter commented on app/Models/Signup.php:
```diff
@@ -197,29 +197,13 @@ public function toQuasarPayload()
-            }
-
-            return $this->quantity;
-        }
```
I was just [asking about trashing all of this](https://dosomething.slack.com/archives/C0YGXUE01/p1575329708161500) as I was looking into aggregating quantity over in #964.

DFurnes replied:
Ah, a good coincidence! 💥

aaronschachter commented:
Re: saving `quantity` / `quantity_pending` on the Signup -- in the pre-Phoenix-Next days, Gambit used to have default messaging that could work for any campaign and we'd use the fields for a user's photo submission confirmation: e.g. `Great, we've got you down for {{quantity_pending}} {{rb_noun}} {{rb_verb}}. If you've got more {{rb_noun}} {{rb_verb}}, text START to submit another photo`. I believe we removed support for those [tags](https://github.com/DoSomething/gambit-admin/wiki/Tags) long ago -- so I'd imagine we're okay to remove them.

DFurnes commented:
Oh interesting – it turns out we're updating the `quantity` on the post [when we create it](https://github.com/DoSomething/rogue/blob/7862db45ba1e8770e8736dee071e85a11426a17e/app/Repositories/PostRepository.php#L134) or [update it](https://github.com/DoSomething/rogue/blob/c91c291076697c262b40c9fca4666286e79dd383/app/Repositories/PostRepository.php#L155), so we might be able to simply return it from the database rather than re-calculate it on each "read".

I'd be a teeny bit worried that there are ways we modifying posts _outside_ of the `POST v3/posts` endpoint though (such as running scripts or the `DELETE v3/posts`) that could cause this database field to get out of sync though... if we do need this, we should consider making it a model event.

DFurnes commented:
> Re: saving `quantity` / `quantity_pending` on the Signup -- in the pre-Phoenix-Next days, Gambit used to have default messaging that could work for any campaign and we'd use the fields for a user's photo submission confirmation: e.g. `Great, we've got you down for {{quantity_pending}} {{rb_noun}} {{rb_verb}}. If you've got more {{rb_noun}} {{rb_verb}}, text START to submit another photo`. I believe we removed support for those [tags](https://github.com/DoSomething/gambit-admin/wiki/Tags) long ago -- so I'd imagine we're okay to remove them.

Iiiinteresting! Yeah, I also vaguely remember the idea of admins "approving" changes in quantity which I think got scrapped along the way.

katiecrane commented:
@DFurnes in terms of admins approving quantity, I think that kind of implicitly happens now that posts have distinct quantities. If we needed to know the "approved" quantity, we could sum up the quantity of all the approved posts.

As you found, the `quantity` on the Signup does get updated when a new post comes in! I'm VERY in favor of making that trigger as a model event (which I kind of thought it already did...) - so we'll want to keep that. As far as I can tell, `quantity_pending` is no longer used at all, let's trash it!

DFurnes commented:
Thanks @aaronschachter and @katiecrane! I've updated this PR to read quantity directly from the appropriate database columns in d2fbe377 & update `signups.quantity` via a model event 
in ac939bb. Finally, I dropped `quantity_pending` in c89b806e.

katiecrane commented on app/Models/Signup.php:
```diff
@@ -31,7 +31,7 @@ class Signup extends Model
      * @var array
      */
-    protected $fillable = ['id', 'northstar_id', 'campaign_id', 'quantity', 'quantity_pending', 'why_participated', 'source', 'source_details', 'details', 'created_at', 'updated_at'];
+    protected $fillable = ['id', 'northstar_id', 'campaign_id', 'why_participated', 'source', 'source_details', 'details', 'created_at', 'updated_at'];
```
Should `quantity` be fillable still?

DFurnes replied:
Nope – we don't want anyone to set quantity on the post by hand (only via the model event).

katiecrane replied:
Did you mean on the signup?

DFurnes replied:
Oh yep - sorry! 🤦‍♂ We do want folks to set quantity on the posts, but the field on the signup should not be mass-assignable & only updated by this model event.

katiecrane commented on tests/Http/SignupTest.php:
```diff
@@ -642,46 +642,6 @@ public function testSignupsIndexAsAdminWithFilters()
-        $this->assertEquals(1, $decodedResponse['data'][0]['quantity']);
-        $this->assertEquals(2, $decodedResponse['data'][1]['quantity']);
-    }
-
```
Is this functionality not something we do anymore?

DFurnes replied:
I don't think so (I believe it was added for Domino), and it's a little awkward to write out the test since this is calculated based on the posts. We could update this to sort by a different field if we want coverage for this behavior.

katiecrane replied:
Oh DOMINO blast from the past!

katiecrane replied:
So does this `orderBy` parameter not work anymore? I don't remember seeing anything about that in this PR so I'm having trouble wrapping my head around this

DFurnes replied:
It does! I had removed this test when I originally thought we weren't storing `quantity` on the post anymore, but should probably re-add it to keep the test coverage.

DFurnes replied:
Re-added this test in f0a485c.
